apps/src/netsim/NetSimLobby.js

```diff
@@ -592,13 +592,19 @@ NetSimLobby.prototype.buildShardChoiceList_ = function (
 
 /**
  * Generate a unique shard key from the given seed
- * @param {string} seed
+ * @param {string} seed - either the section ID, random UUID, or shared shard seed.
  * @private
  */
 NetSimLobby.prototype.makeShardIDFromSeed_ = function (seed) {
-  // md5 returns a 32-character hex string so these should always be 35-character strings
-  // Previously we truncated to 48 characters
-  return 'ns_' + md5(this.levelKey_ + '_' + seed);
+  const MAX_LENGTH = 48;
+  const NET_SIM_PREFIX = 'ns-';
```

Comment:
I updated underscore to dash since we perform a regex check for `_(\d+)` to find section ID in shard ID in `doesUserOwnShard`: https://github.com/code-dot-org/code-dot-org/blob/cce3b56e4931d8b7fb15d62e81894d62fe85b6d2/apps/src/netsim/NetSimUtils.js#L424-L428

Reply:
FWIW, I'm not actually sure this matters so long as the last underscore before the regex remains in tact.

It was an underscore before: https://github.com/code-dot-org/code-dot-org/blame/a80bc179224ae91e12ba79bac3cf2e60426af6dd/apps/src/netsim/NetSimLobby.js#L598

I think the regex /_(\d+)$/ is looking for the underscore before the numeric section ID at the end, and earlier underscores are ignored.

Reply:
Yeah - I was thinking of the case if the hash happened to begin with a digit.